bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateService.java

```diff
@@ -199,6 +199,17 @@ public Void call() {
             }
         });
     }
+    
+    public void cancelFirmwareUpdate(final ThingUID thingUID){
+        Preconditions.checkNotNull(thingUID, "Thing UID must not be null.");
+        final FirmwareUpdateHandler firmwareUpdateHandler = getFirmwareUpdateHandler(thingUID);
+        if (firmwareUpdateHandler == null) {
+            throw new IllegalArgumentException(
+                    String.format("There is no firmware update handler for thing with UID %s.", thingUID));
+        }
+        logger.debug("Canceling firmware update for thing with UID {}.", thingUID);
+        firmwareUpdateHandler.cancel();
```

Comment:
> how do we inform callers that the update has been successfully canceled?

FirmwareUpdateHandler calls ProgressCallback.failed() if a update was canceled. This results in a FirmwareUpdateResultInfo event.